dango/dex/src/core/liquidity_pool.rs

```diff
@@ -254,6 +306,98 @@ impl PassiveLiquidityPool for PairParams {
 
         Ok((reserve, input))
     }
+
+    fn reflect_curve(
+        &self,
+        base_denom: Denom,
+        quote_denom: Denom,
+        reserves: &CoinPair,
+    ) -> StdResult<(
+        impl Iterator<Item = StdResult<(Udec128, Uint128)>>,
+        impl Iterator<Item = StdResult<(Udec128, Uint128)>>,
+    )> {
+        let base_reserves = reserves.amount_of(&base_denom)?;
+        let quote_reserves = reserves.amount_of(&quote_denom)?;
+        let spot_price = self.spot_price(&base_denom, &quote_denom, reserves)?;
+
+        let swap_fee_rate = self.swap_fee_rate.into_inner();
+
+        // Construct the bid order iterator.
+        let mut bid_i = 1;
+        let mut bid_prev_amount = Uint128::ZERO;
+        let mut bid_price = spot_price.checked_mul(Udec128::ONE.checked_sub(swap_fee_rate)?)?;
```

Comment:
@pacmanifold I think this should be half the swap fee rate. You have half on the buy side, half on the sell side, combined the spread equals the swap fee rate.